apps/nextjs/src/app/api/admin/moderation/[moderationId]/invalidate/route.ts

```diff
@@ -0,0 +1,83 @@
+import { auth, clerkClient } from "@clerk/nextjs/server";
+import { Moderations, SafetyViolations } from "@oakai/core";
+import { prisma } from "@oakai/db";
+import { NextRequest } from "next/server";
+
+async function handler(
+  req: NextRequest,
+  { params }: { params: { moderationId: string } },
+): Promise<Response> {
+  const moderationId = params.moderationId;
+
+  if (!moderationId) {
+    return new Response("Invalid moderationId", {
+      status: 400,
+    });
+  }
+
+  const { userId } = auth();
+  if (!userId) {
+    return new Response("Unauthorized", {
+      status: 401,
+    });
+  }
+
+  const userIsAuthorised = await isAuthorised({ userId });
+  if (!userIsAuthorised) {
+    return new Response("Unauthorized", {
+      status: 401,
+    });
+  }
+
+  try {
+    await invalidateToxicModerationAndRemoveSafetyViolation({
+      moderationId,
+      invalidatedBy: userId,
+    });
+  } catch (error) {
+    return new Response(
+      `Failed to invalidate moderation: ${error?.["message"]}`,
+      {
+        status: 500,
+      },
+    );
+  }
+
+  return new Response("Moderation invalidated and safety violation removed", {
+    status: 200,
+  });
+}
+
+export const GET = handler;
```

Comment:
Traditionally more of a POST kinda thing if strictly REST, but as we want people to be able to just click it from a slack message, GET made sense

Reply:
Is there any chance a slack message ends up being sent as an email to anyone? If so, the mail client might fetch the URL to see if it is valid / anti-spam

Reply:
Might be better to go to a page that has a form on it if so